File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        
        sol = []
        cur = 1

        zero_idx = None
        if nums[0] == 0:
            zero_idx = 0
        
        for i in range(1, len(nums)):
            if nums[i] == 0 and zero_idx == None:
                zero_idx = i
                cur *= nums[0]
            else:
                cur *= nums[i]

        if zero_idx != None:
            sol = [0] * len(nums)
            sol[zero_idx] = cur
            return sol
        
        sol.append(cur)
        
        for i in range(1, len(nums)):
            cur //= nums[i]
            cur *= nums[i - 1]
            sol.append(cur)
            print(cur)
        
        return sol

File: Data Structures & Algorithms/two-integer-sum/submission-0.cpp
#include <vector>
#include <unordered_map>
using namespace std;
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> sol;
        unordered_map<int, int> seen;
        for(int i = 0; i < nums.size(); i++) {
            if(seen.contains(target - nums[i])) {
                sol.push_back(seen[target - nums[i]]);
                sol.push_back(i);
                break;
            }
            else {
                seen[nums[i]] = i;
            }
        }
        return sol;
    }
};
